sway/config/output.c

```diff
@@ -377,6 +377,21 @@ static const uint32_t *bit_depth_preferences[] = {
 	},
 };
 
+const char *sway_render_format_to_string(uint32_t render_format) {
```

Comment:
I need we need to decide on a more robust way of printing all the formats. There are certainly common formats not filled out in this table. One solution to this problem is introducing a wlroots function to stringify this because it keeps track of all supported formats in a table it internally manages.

Reply:
Maybe in the future printing the format modifier as well.

Reply:
Oh nevermind, libdrm already gives us `drmGetFormatName` and `drmGetFormatModifierName`!

Reply:
Neat, I briefly looked for that but got sidetracked and went with this approach as a quick stopgap.

One question: looks like this was introduced in `drm` just under a year ago, do we need to worry about supporting versions of `drm` prior to its introduction? Or do new sway releases typically require fairly recent `drm` versions?

Reply:
No, wlroots uses this funciton. If someone were to try to build sway on these old versions, wlroots as a dependency wouldn't even compile.

Reply:
There is no need to try to get this working with every existing format. The only formats which can be picked here are the ones passed in by Sway to wlroots by `wlr_output.render_format`. Sway has a fixed list of bit depth → format combinations, right above.

Also note, the libdrm functions don't return the same string as the `#define`: it returns the four character code. libdrm returns e.g. AR24 for `DRM_FORMAT_ARGB8888`.

I do not know whether this string is a good value to expose.

Reply:
> There is no need to try to get this working with every existing format. The only formats which can be picked here are the ones passed in by Sway to wlroots via wlr_output.render_format. Sway has a fixed list of bit depth → format combinations, right above.

Agreed, but if a generic method exists it would be nice to use.

> Also note, the libdrm functions don't return the same string as the #define: it returns the four character code. libdrm returns e.g. AR24 for DRM_FORMAT_ARGB8888.

I noticed that after I commented, and personally would prefer the more verbose `ARGB8888` vs. `AR24`.

> I do not know whether this string is a good value to expose.

Is this in reference to `AR24` or the render format in general? If the latter, would you prefer just the bit depth or a different value?

Reply:
> Is this in reference to AR24 or the render format in general?

A string representing the render format might tie out hands in the future...